backend/src/lib.rs

```diff
@@ -1,104 +1,176 @@
+#[cfg(feature = "halo2")]
+mod halo2_structs;
+mod pilcom_cli;
+
 use ast::analyzed::Analyzed;
-use number::FieldElement;
-use std::io;
+use number::{DegreeType, FieldElement};
+use std::{io, marker::PhantomData, path::Path};
 use strum::{Display, EnumString, EnumVariantNames};
 
 #[derive(Clone, EnumString, EnumVariantNames, Display)]
-pub enum Backend {
+pub enum BackendType {
     #[cfg(feature = "halo2")]
     #[strum(serialize = "halo2")]
     Halo2,
     #[cfg(feature = "halo2")]
-    #[strum(serialize = "halo2-aggr")]
-    Halo2Aggr,
-    #[cfg(feature = "halo2")]
     #[strum(serialize = "halo2-mock")]
     Halo2Mock,
-    // At the moment, this enum is empty without halo2, but it is always built
-    // as part of the infrastructure to eventually support other backends.
+    #[strum(serialize = "pilcom-cli")]
+    PilcomCli,
 }
 
-/// Create a proof for a given PIL, fixed column values and witness column values
-/// using the chosen backend.
+impl BackendType {
+    pub fn build<T: FieldElement>(&self) -> &'static dyn BackendFactory<T> {
+        #[cfg(feature = "halo2")]
+        const HALO2_FACTORY: WithSetupFactory<halo2::Halo2Prover> = WithSetupFactory(PhantomData);
```

Comment:
why does this one use `halo2::...` and Mock uses `halo2_structs`?

Reply:
Because `Halo2Prover` is defined in the `halo2` crate, and the mock is defined in the module `halo2_structs` of the `backend` crate that makes use of the `halo2` crate.